### PR TITLE
add a Dockerfile for creating nim-waku images

### PR DESCRIPTION
**Pull request by jakubgs**
Part of work to deploy `nim-waku` to its own fleet: https://github.com/status-im/infra-nimbus/issues/13
Infra repo: https://github.com/status-im/infra-nim-waku

Images are uploaded to our Docker Hub account:
https://hub.docker.com/r/statusteam/nim-waku

It Just Werks™:
```
 > docker images statusteam/nim-waku          
REPOSITORY            TAG          IMAGE ID         CREATED          SIZE
statusteam/nim-waku   latest       5cb78b4e0400     5 minutes ago    37.2MB

 > docker run --rm -it statusteam/nim-waku:latest  
Usage: 

wakunode [OPTIONS]...

The following options are available:

 --log-level               Sets the log level.
 --tcp-port                TCP listening port.
 --udp-port                UDP listening port.
...
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# BUILD IMAGE --------------------------------------------------------
+FROM nimlang/nim:1.2.0-alpine AS nim-build
```
Comment (kdeme) [inline Dockerfile:2]: The nimlang version should not be needed.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+# Build the actual node binary
+RUN make wakunode
+
+# ACTUAL IMAGE -------------------------------------------------------
+FROM alpine:3.11
+
+LABEL maintainer="jakub@status.im"
+LABEL source="https://github.com/status-im/nim-waku"
+LABEL description="nim-waku is a Waku protocol node written in Nim"
```
Comment (kdeme) [inline Dockerfile:22]: Perhaps use the original description I had? As it is really a node that talks waku but e.g. also still Whisper.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+RUN mkdir /app
+WORKDIR /app
+ADD . .
+
+# Ran separately from 'make' to avoid re-doing
+RUN git submodule update --init --recursive
+
+# Build the actual node binary
+RUN make wakunode
```
Comment (kdeme) [inline Dockerfile:15]: If we want to use metrics, we have to add `NIMFLAGS="-d:insecure"` here

Comment (kdeme) [inline Dockerfile:15]: Also, I believe part of Nim compilation does benefit here from `-j$(nproc)`

File: Dockerfile (hunk 1 above)
Comment (jakubgs) [reply Dockerfile:2]: Why is locking a specific Nim version a bad idea?

File: Dockerfile (hunk 2 above)
Comment (jakubgs) [reply Dockerfile:22]: Sure.

File: Dockerfile (hunk 3 above)
Comment (jakubgs) [reply Dockerfile:15]: Sure.

Comment (jakubgs) [reply Dockerfile:15]: That's better if it's parametrized.

File: Makefile [hunk 4]
```diff
@@ -10,6 +10,12 @@ SHELL := bash # the shell used internally by Make
 # used inside the included makefiles
 BUILD_SYSTEM_DIR := vendor/nimbus-build-system
 
+# Docker image name
+DOCKER_IMAGE_TAG ?= latest
+DOCKER_IMAGE_NAME ?= statusteam/nim-waku:$(DOCKER_IMAGE_TAG)
```
Comment (oskarth) [inline Makefile:15]: Can we clarify that this is nim-waku v1? Otherwise it might get confusing when we have experimental nim-waku v2 images.

Comment (jakubgs) [reply Makefile:15]: That's what tags are for. When you want to switch to v2 you'll just change the tag.

Comment (kdeme) [reply Makefile:15]: Yeah, when the time is there, we can add a make target for v2 and add a different tag.

Comment (oskarth) [reply Makefile:15]: couldn't the tag be 1 then? :) latest to me implies it could be either

anyway not a big deal

Comment (jakubgs) [reply Makefile:15]: Feel free to change it, I'll be using one named `deploy-test` for the fleet anyway.

**Pull request merged**